src/bun.js/bindings/webcore/ReadableStream.cpp

```diff
@@ -136,8 +136,16 @@ JSC::JSValue ReadableStream::tee()
     arguments.append(JSC::jsBoolean(true));
     ASSERT(!arguments.hasOverflowed());
     auto returnedValue = invokeReadableStreamFunction(lexicalGlobalObject, privateName, JSC::jsUndefined(), arguments);
+
+    JSArray* array = JSC::jsDynamicCast<JSArray*>(returnedValue.value_or(JSC::jsUndefined()));
+
+    VM& vm = lexicalGlobalObject.vm();
+    JSC::JSObject* object = JSC::constructEmptyObject(&lexicalGlobalObject, lexicalGlobalObject.objectPrototype(), 2);
+
+    object->putDirect(vm, JSC::Identifier::fromString(vm, "branch1"_s), array->getDirectIndex(&lexicalGlobalObject, 0), 0);
```

Comment:
We can skip creating this object

Instead, either return 
```c++
EncodedJSValue[2]{  JSValue::encode(array->getDirectIndex(&lexicalGlobalObject, 0)), JSValue::encode(array->getDirectIndex(&lexicalGlobalObject, 0)) };
```

or make it a pointer we pass to this function

Reply:
then how to get value in Zig part, still using arrayIterator?